WordPress/Classes/ViewRelated/Stats/Insights/SiteStatsInsightsViewModel.swift

```diff
@@ -367,73 +381,72 @@ private extension SiteStatsInsightsViewModel {
     }
 
     func createTagsAndCategoriesRows() -> [StatsTotalRowData] {
-        let tagsAndCategories = store.getTopTagsAndCategories()
-        return tagsAndCategories?.map { StatsTotalRowData.init(name: $0.label,
-                                                               data: $0.value.displayString(),
-                                                               dataBarPercent: StatsDataHelper.dataBarPercentForRow($0, relativeToRow: tagsAndCategories?.first),
-                                                               icon: tagsAndCategoriesIconForItem($0),
-                                                               showDisclosure: true,
-                                                               disclosureURL: StatsDataHelper.disclosureUrlForItem($0),
-                                                               childRows: childRowsForItem($0)) }
-            ?? [StatsTotalRowData]()
-    }
+        guard let tagsAndCategories = store.getTopTagsAndCategories()?.topTagsAndCategories else {
+            return []
+        }
 
-    func tagsAndCategoriesIconForItem(_ item: StatsItem) -> UIImage? {
 
-        if let children = item.children,
-            children.count > 0 {
-            return Style.imageForGridiconType(.folderMultiple)
+        return tagsAndCategories.map {
+            let viewsCount = $0.viewsCount ?? 0
+
+            return StatsTotalRowData(name: $0.name,
+                                     data: viewsCount.abbreviatedString(),
+                                     dataBarPercent: Float(viewsCount) / Float(tagsAndCategories.first?.viewsCount ?? 1),
+                                     icon: tagsAndCategoriesIconForKind($0.kind),
+                                     showDisclosure: true,
+                                     disclosureURL: $0.url,
+                                     childRows: childRowsForItems($0.children))
+
         }
+    }
 
-        switch item.alternateIconValue {
-        case "category":
+    func tagsAndCategoriesIconForKind(_ kind: StatsTagAndCategory.Kind) -> UIImage? {
+        switch kind {
+        case .folder:
+            return Style.imageForGridiconType(.folderMultiple)
+        case .category:
             return Style.imageForGridiconType(.folder)
-        default:
+        case .tag:
             return Style.imageForGridiconType(.tag)
         }
     }
 
-    func childRowsForItem(_ item: StatsItem) -> [StatsTotalRowData] {
-
-        guard let children = item.children as? [StatsItem] else {
-            return [StatsTotalRowData]()
+    func childRowsForItems(_ children: [StatsTagAndCategory]) -> [StatsTotalRowData] {
+        return children.map {
+            StatsTotalRowData.init(name: $0.name,
+                                   data: "",
```

Comment:
I see you removed this:
`data: ($0.value != nil) ? $0.value.displayString() : ""`

That was a placeholder so that when a Tag/Category child had a value, it would show. It's not actually meant to be an empty value. Is that still on your todo list (provide values for children)?

Reply:
I don't think those specific insights ever have value — the JSON returned by the API certainly doesn't have it, and looking at Calypso there aren't any there either.

Reply:
Gotcha. I'll bring it up to @SylvesterWilmott so he's aware we don't have this data. Thanks!